book/012-vis-adv.Rmd

```diff
@@ -1 +1,155 @@
 # {ggplot2} advanced {#vis-adv}
+
+## Facetting
+
+<details><summary>*Click here to show setup code.*</summary>
+```{r setup-vis-adv}
+library(tidyverse)
+library(ggsci)
+library(ggpubr)
+library(patchwork)
+```
+</details>
+
+<br>
+
+"Facetting" (or trellis plots, lattice plots) denotes an idea of dividing a graphic into sub-graphics based on the (categorical) values of one or more variables of a dataset.
+
+```{r, eval=FALSE}
+facet_grid(facets = . ~ ., scales = "fixed", ...)
+facet_wrap(facets = <variable> ~ ., scales = "fixed", ...)
+```
+
+**facet**: Formula with splitting variable.
+
+**scales**: Scale of the axes over the sub-graphics.
+
+The position of `<variable>` denotes on which axis the facets will be shown:
+
+- `<variable> ~ .` `r icon::fa_arrow_right()` y-axis
+
+- `. ~ <variable> ``r icon::fa_arrow_right()` x-axis
+
+![](img/facets.png)
+
+### `facet_wrap()`
+
+```{r, fig.width=8, fig.height=4}
+facet_x = ggplot(mpg, aes(displ, hwy)) +
+  geom_point(aes(colour = fl)) +
+  scale_color_nejm() +
+  facet_grid(. ~ year) +
+  theme_pubr()
```

Comment:
I want to showcase what happens if the `.` is on a specific side.

Reply:
The question is about `theme_pubr()` .

Reply:
Ah didn't know since you were highlighting 4 lines.

I use `theme_pubr()` consistently throughout all plots. Some look really bad with the default theme. Not just in a subjective manner but making certain elements almost unreadable (e.g. text, alpha point settings). 

Potential for confusion: Well, with any line being added the confusion potential increases. But would this be different with even a built-in call to `theme_minimal()`? I know there is currently no section covering themes and `theme()` in particular but the function name per se should be descriptive enough?

Or did you refer to something different in terms of confusion potential?

Reply:
I remember highlighting one line.

The plotting code contains five lines, two of which are noise. This may cause confusion. I think theming should be specified in one place so that it is applied consistently to all plots. There is `52-theme` which mentions that, not included in the overall text yet.

If the plots look suboptimal, this is a good motivation for introducing theming later.

So far no code in `master` contains theming (except for `013-vis-ext.Rmd` which I missed). I like that.

Reply:
> If the plots look suboptimal, this is a good motivation for introducing theming later.

If then earlier I would say - to prevent having the majority of the book containing suboptimal themed plots.

I'll take a look where I can put in a small section about theming. I am a bit worried about the time and remaining exercises that need to be squeezed in.